api/v1alpha1/reaper_types.go

```diff
@@ -68,6 +68,8 @@ type AuthProvider struct {
 	Username string `json:"username,omitempty" yaml:"username,omitempty"`
 
 	Password string `json:"password,omitempty" yaml:"password,omitempty"`
+
+	SecretRef string `json:"secretKeyName,omitempty" yaml:"secretKeyName,omitempty"`
```

Comment:
There are a couple things here. First, can we rename `SecretRef` to `CassandraUserSecretName`? That is more consistent with `JmxUserSecretName`. 

Secondly, I think we can rid of `AuthProvider`. It was part of my initial prototypes of reaper-operator where I was simply started out by copying some of the configs specified at http://cassandra-reaper.io/docs/configuration/backend_specific/. We should be able to remove the struct and move the secret field.

Reply:
Yeah, removing AuthProvider sounds fine.

Reply:
Or should we push both, the imx and Cassandra auth under one struct? Instead of both being some string types in the upper level?